src/tokens/erc1155.ts

```diff
@@ -0,0 +1,65 @@
+import fetchQuestActionParams from '../quests/fetchQuestData.js'
+import axios from 'axios'
+import { type Address, type PublicClient } from 'viem'
+
+/**
+ * Fetches ERC1155 token metadata from a given contract.
+ * @async
+ * @param {PublicClient} client The Viem client instance.
+ * @param {string} contractAddress The ERC1155 contract address.
+ * @param {number} tokenId The token ID.
+ * @returns {Promise<any>} The metadata of the ERC1155 token.
+ */
+export async function fetchERC1155Metadata(
+  client: PublicClient,
+  contractAddress: string,
+  tokenId: number,
+): Promise<any> {
+  const tokenURI: string = await (client.readContract({
+    address: contractAddress as Address,
+    abi: [
+      'function uri(uint256 tokenId) external view returns (string memory)',
+    ],
+    functionName: 'uri',
+    args: [tokenId],
+  }) as Promise<string>)
+
+  const response = await axios.get(tokenURI)
+  return response.data
+}
+
+/**
+ * Fetches the media URL from ERC1155 token metadata and performs basic validation or sanitization.
```

Comment:
Don't think there's "basic validation or sanitization"

Reply:
There's not - I'll change this comment we discussed using a service for this but I think we're just going to handle it with the dom purify in the front-end

Reply:
Might be worth purifying on the BE wherever we serve the info to prevent passing dangerous data to third party clients or w/e